project-base/storefront/hooks/cart/useReloadCart.ts

```diff
@@ -4,23 +4,31 @@ import { getUrlWithoutGetParameters } from 'helpers/parsing/getUrlWithoutGetPara
 import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
 import { useCurrentUserData } from 'hooks/user/useCurrentUserData';
 import { useRouter } from 'next/router';
-import { useEffect, useMemo } from 'react';
+import { useEffect } from 'react';
 import { usePersistStore } from 'store/zustand/usePersistStore';
+import { useClient } from 'urql';
+import { CartQueryDocumentApi } from 'graphql/generated';
 
 export const useReloadCart = (): void => {
-    const { modifications, refetchCart } = useCurrentCart(false);
+    const { modifications } = useCurrentCart(false);
     const [changePaymentInCart] = useChangePaymentInCart();
     const t = useTypedTranslationFunction();
     const router = useRouter();
-    const slug = useMemo(() => getUrlWithoutGetParameters(router.asPath), [router.asPath]);
+    const slug = getUrlWithoutGetParameters(router.asPath);
     const { isUserLoggedIn } = useCurrentUserData();
+    const updateLastVisitedSlug = usePersistStore((store) => store.updateLastVisitedSlug);
+    const lastVisitedSlug = usePersistStore((store) => store.lastVisitedSlug);
     const cartUuid = usePersistStore((store) => store.cartUuid);
+    const client = useClient();
 
     useEffect(() => {
-        if (cartUuid || isUserLoggedIn) {
-            refetchCart();
+        const hasSlugChanged = lastVisitedSlug !== slug;
+        updateLastVisitedSlug(slug);
```

Comment:
This seems like something is not done correctly on the way. Cannot this be done better way so it is reacting simply for slug change?

Reply:
I think we can. I dont know why it has stayed here.